docs/general/run-a-node/maintenance-guides/handling-network-upgrades.md

```diff
@@ -1,81 +1,173 @@
+  consensus state or changing the genesis document.
+- *Dump & restore upgrade* requires that you wipe the state and run a new
+  version of the Oasis Core from a fresh genesis document.
 
-Before an upgrade we will update the [Upgrade Log](../upgrade-log.md) to specify the epoch height at which the upgrade will take place. Additionally, an upgrade descriptor will be provided which can be used to instruct the node to shutdown and export state at the start of the upgrade epoch.
+## Stop the node
 
-To submit the provided upgrade descriptor use the following command:
+The upgrade proposal transaction includes the *upgrade height* field. This is
+the epoch at which, once reached, your node will automatically stop, writing
+the following in the log
 
-```bash
-oasis-node control upgrade-binary \
- -a unix:/serverdir/node/internal.sock \
- <PATH-TO-UPGRADE-DESCRIPTOR.json>
+```text
+{"caller":"mux.go:426","level":"debug","module":"abci-mux","msg":"dispatching halt hooks for upgrade","ts":"2022-05-06T13:11:41.721994647Z"}
 ```
 
-To verify that the upgrade descriptor has been received, grep your node's logs for "received upgrade descriptor" message, e.g.:
-
-> {"caller":"upgrade.go:60","epoch":5102,"level":"info","module":"upgrade","msg":"received upgrade descriptor, scheduling shutdown","name":"testnet-upgrade-2021-03-24","ts":"2021-03-24T09:56:52.944552808Z"}
+and on the error output:
 
-If for any reason you would need to cancel a scheduled pending upgrade, use the `cancel-upgrade` command.
+```text
+panic: upgrade: reached upgrade epoch
+```
 
-Node reaching the upgrade epoch will automatically export network state to a genesis file under the following path:`exports/genesis-<CHAIN-ID>-at-<DUMP-BLOCK-HEIGHT>.json`.
+Then, the `oasis-node genesis dump` command will automatically execute and
+the consensus state will be written into
+`genesis-<CHAIN_ID>-at-<LATEST_HEIGHT>.json` file in the working directory,
+where `CHAIN_ID` is the chain ID of the network and `LATEST_HEIGHT` is the
+round of the last block before the upgrade epoch. This command, depending on the
+size of the state, may take some time to finish.
 
-## Manually exporting network state
+:::info
 
-To dump the state of the network to a genesis file, run:
+To check the network's current height and epoch, run:
 
 ```bash
-oasis-node genesis dump \
-  -a unix:/serverdir/node/internal.sock \
-  --genesis.file /serverdir/etc/genesis_dump.json \
-  --height <HEIGHT-TO-DUMP>
+oasis-node control status -a unix:/serverdir/node/internal.sock
 ```
 
-replacing `<HEIGHT-TO-DUMP>` with the block height we specified.
+and observe the value of the `consensus.latest_height` and
+`consensus.latest_epoch` fields respectively.
 
-:::caution
+:::
 
-You can only run the following command _after_ the `<HEIGHT-TO-DUMP>` block height has been reached on the network.
+## Upgrade Oasis Core
 
-To see the network's current height, run:
+Make sure you upgrade your [Oasis Core](../prerequisites/oasis-node.md)
```

Comment:
I would remove this `Upgrade Oasis Core` part.

The accompanying upgrade log should/will have instructions on what to do once the upgrade epoch is reached and the node has stopped.